test/boost/cql_query_test.cc

```diff
@@ -4313,3 +4315,207 @@ SEASTAR_TEST_CASE(test_rf_expand) {
+            BOOST_CHECK_EQUAL(normalize_white_space(ss.str()), normalize_white_space(ct.second));
+        }
+    });
+}
+
+SEASTAR_TEST_CASE(test_describe_view_schema) {
+    return do_with_cql_env_thread([] (cql_test_env& e) {
+        std::string base_table = "CREATE TABLE ks.cf (\n"
+                "    pk blob,\n"
+                "    pk1 blob,\n"
+                "    ck blob,\n"
+                "    ck1 blob,\n"
+                "    \"COL1\" blob,\n"
+                "    col2 blob,\n"
+                "    PRIMARY KEY ((pk, pk1), ck, ck1)\n"
+                ") WITH CLUSTERING ORDER BY (ck ASC, ck1 ASC)\n"
+                "    AND bloom_filter_fp_chance = 0.01\n"
+                "    AND caching = {'keys': 'ALL','rows_per_partition': 'ALL'}\n"
+                "    AND comment = ''\n"
+                "    AND compaction = {'class': 'SizeTieredCompactionStrategy'}\n"
+                "    AND compression = {'sstable_compression': 'org.apache.cassandra.io.compress.LZ4Compressor'}\n"
+                "    AND crc_check_chance = 1\n"
+                "    AND dclocal_read_repair_chance = 0.1\n"
+                "    AND default_time_to_live = 0\n"
+                "    AND gc_grace_seconds = 864000\n"
+                "    AND max_index_interval = 2048\n"
+                "    AND memtable_flush_period_in_ms = 0\n"
+                "    AND min_index_interval = 128\n"
+                "    AND read_repair_chance = 0\n"
+                "    AND speculative_retry = '99.0PERCENTILE';\n";
+
+        std::unordered_map<std::string, std::string> cql_create_tables {
+          {"cf_view", "CREATE MATERIALIZED VIEW ks.cf_view AS\n"
+              "    SELECT \"COL1\", pk, pk1, ck1, ck\n"
+              "    FROM cf\n"
+              "    WHERE pk IS NOT null AND \"COL1\" IS NOT null AND pk1 IS NOT null AND ck1 IS NOT null AND ck IS NOT null\n"
+              "    PRIMARY KEY (\"COL1\", pk, pk1, ck1, ck)\n"
+              "    WITH CLUSTERING ORDER BY (pk ASC, pk1 ASC, ck1 ASC, ck ASC)\n"
+              "    AND bloom_filter_fp_chance = 0.01\n"
+              "    AND caching = {'keys': 'ALL','rows_per_partition': 'ALL'}\n"
+              "    AND comment = ''\n"
+              "    AND compaction = {'class': 'SizeTieredCompactionStrategy'}\n"
+              "    AND compression = {'sstable_compression': 'org.apache.cassandra.io.compress.LZ4Compressor'}\n"
+              "    AND crc_check_chance = 1\n"
+              "    AND dclocal_read_repair_chance = 0.1\n"
+              "    AND default_time_to_live = 0\n"
+              "    AND gc_grace_seconds = 864000\n"
+              "    AND max_index_interval = 2048\n"
+              "    AND memtable_flush_period_in_ms = 0\n"
+              "    AND min_index_interval = 128\n"
+              "    AND read_repair_chance = 0\n"
+              "    AND speculative_retry = '99.0PERCENTILE';\n"},
+          {"cf_index_index", "CREATE INDEX cf_index ON ks.cf(col2);"},
```

Comment:
Would also be good to check three more indexes: ks.cf(pk), ks.cf(pk1) and ks.cf(ck) - because it's possible to index also a key column (I think your code is correct in this case - I don't suspect a bug - but it wouldn't hurt to check, and it should be very easy with your code that can easily add more tests in this looo).